posthog/clickhouse/cancel.py

```diff
@@ -2,13 +2,44 @@
 
 from posthog.api.services.query import logger
 from posthog.clickhouse.client import sync_execute
+from posthog.clickhouse.client.connection import default_client
 from posthog.settings import CLICKHOUSE_CLUSTER
 
 
 def cancel_query_on_cluster(team_id: int, client_query_id: str) -> None:
-    result = sync_execute(
-        f"KILL QUERY ON CLUSTER '{CLICKHOUSE_CLUSTER}' WHERE query_id LIKE %(client_query_id)s",
-        {"client_query_id": f"{team_id}_{client_query_id}%"},
-    )
+    initiator_host = None
+    failed_to_find_initiator_host = False
+    try:
+        result = sync_execute(
+            f"""
+            SELECT hostname()
+            FROM clusterAllReplicas(posthog, system.query_log)
+            WHERE query_id LIKE %(client_query_id)s AND (event_time > (now() - toIntervalDay(1)))
```

Comment:
I don't think we will kill queries running for more than a day so I reduced the search to the last day.

Reply:
I'm finally using `system.processes` instead of the query_log, which makes way more sense for the use case.